src/Core/Billing/Models/MaskedPaymentMethodDTO.cs

```diff
@@ -0,0 +1,160 @@
+                {
+                    return new MaskedPaymentMethodDTO(
+                        PaymentMethodType.BankAccount,
+                        $"{bankAccount.BankName}, *{bankAccount.Last4}",
+                        false);
+                }
+            default:
+                {
+                    return null;
+                }
+        }
+    }
+
+    private static MaskedPaymentMethodDTO FromStripeBankAccountPaymentMethod(Stripe.PaymentMethodUsBankAccount bankAccount)
+    {
+        var description = $"{bankAccount.BankName}, *{bankAccount.Last4}";
+
+        return new MaskedPaymentMethodDTO(
+            PaymentMethodType.BankAccount,
+            description,
+            false);
+    }
+
+    private static MaskedPaymentMethodDTO FromStripeCardPaymentMethod(Stripe.PaymentMethodCard card)
+        => new(
+            PaymentMethodType.Card,
+            GetCardDescription(card.Brand, card.Last4, card.ExpMonth, card.ExpYear),
+            false);
+
+    #region Legacy Source Payments
+    private static MaskedPaymentMethodDTO FromStripeLegacyPaymentSource(Stripe.IPaymentSource paymentSource)
+        => paymentSource switch
+        {
+            Stripe.BankAccount bankAccount => FromStripeBankAccountLegacySource(bankAccount),
+            Stripe.Card card => FromStripeCardLegacySource(card),
+            Stripe.Source { Card: not null } source => FromStripeSourceCardLegacySource(source.Card),
+            _ => null
+        };
+
+    private static MaskedPaymentMethodDTO FromStripeBankAccountLegacySource(Stripe.BankAccount bankAccount)
+    {
+        var status = bankAccount.Status switch
+        {
+            "verified" => "Verified",
+            "errored" => "Invalid",
+            "verification_failed" => "Verification failed",
+            _ => "Unverified"
+        };
+
+        var description = $"{bankAccount.BankName}, *{bankAccount.Last4} - {status}";
+
+        var needsVerification = bankAccount.Status is "new" or "validated";
+
+        return new MaskedPaymentMethodDTO(
+            PaymentMethodType.BankAccount,
+            description,
+            needsVerification);
+    }
+
+    private static MaskedPaymentMethodDTO FromStripeCardLegacySource(Stripe.Card card)
+        => new(
+            PaymentMethodType.Card,
+            GetCardDescription(card.Brand, card.Last4, card.ExpMonth, card.ExpYear),
+            false);
+
+    private static MaskedPaymentMethodDTO FromStripeSourceCardLegacySource(Stripe.SourceCard card)
+        => new(
+            PaymentMethodType.Card,
+            GetCardDescription(card.Brand, card.Last4, card.ExpMonth, card.ExpYear),
+            false);
+    #endregion
+
+    private static string GetCardDescription(
+        string brand,
+        string last4,
+        long expirationMonth,
+        long expirationYear)
+    {
+        var formattedExpirationMonth = string.Concat(expirationMonth < 10 ? "0" : string.Empty, expirationMonth);
```

Comment:
⛏️ Same here. Given an `int`, you can pad the month with zeros to a width of two with `$"...{expirationMonth:00}..."`

Reply:
Updated in https://github.com/bitwarden/server/pull/4140/commits/1b9ac615459d3721c218fa4fb84446386f30d973